File: client/src/dto/actions/CellAttackMetadataDto.ts
export interface CellAttackMetadataDto {
    isRangedAttack: boolean;
    isRetaliated: boolean;
}

export function isCellAttackMetadataDto(
    metadata: unknown
): metadata is CellAttackMetadataDto {
    return (
        typeof metadata === "object" &&
        metadata !== null &&
        "isRangedAttack" in metadata &&
        typeof (metadata as CellAttackMetadataDto).isRangedAttack === "boolean"
        && "isRetaliated" in metadata &&
        typeof (metadata as CellAttackMetadataDto).isRetaliated === "boolean"
    );
}

File: server/dto/actions/cell_attack_metadata_dto.py
from dataclasses import dataclass

from dto.base_dto import BaseDto


@dataclass
class CellAttackMetadataDto(BaseDto):
    isRangedAttack: bool
    isRetaliated: bool

    @staticmethod
    def get_default():
        return CellAttackMetadataDto(isRangedAttack=False, isRetaliated=False)
